### PR TITLE
Methods for managing Association Members

### PR DESCRIPTION
**Pull request by zizouet**
Added the possibility to remove the user from an association.
Added the possibility to change the role of a user in an association.

Enjoy :)))

### CODE REVIEW

File: app/src/main/java/com/github/se/assocify/model/database/UserAPI.kt [hunk 1]
```diff
@@ -328,4 +328,97 @@ class UserAPI(private val db: SupabaseClient, cachePath: Path) : SupabaseApi() {
...
+
+  /**
+   * Removes a user from an association
+   *
+   * @param userId the id of the user to remove
+   * @param associationId the id of the association to remove the user from
+   * @param onSuccess called on success
+   * @param onFailure called on failure
+   */
+  fun removeUserFromAssociation(
```
Comment (Mai-LinhC) [inline app/src/main/java/com/github/se/assocify/model/database/UserAPI.kt:397]: does it delete the line in member_of ?

File: app/src/main/java/com/github/se/assocify/model/database/UserAPI.kt [hunk 2]
```diff
@@ -328,4 +328,97 @@ class UserAPI(private val db: SupabaseClient, cachePath: Path) : SupabaseApi() {
       return PermissionRole(roleId, associationId, type)
     }
   }
+
```
Comment (Mai-LinhC) [inline app/src/main/java/com/github/se/assocify/model/database/UserAPI.kt:331]: no documentation :(

File: app/src/main/java/com/github/se/assocify/model/database/UserAPI.kt (hunk 1 above)
Comment (zizouet) [reply app/src/main/java/com/github/se/assocify/model/database/UserAPI.kt:397]: It deletes everywhere, it's in cascade !

File: app/src/main/java/com/github/se/assocify/model/database/UserAPI.kt (hunk 2 above)
Comment (zizouet) [reply app/src/main/java/com/github/se/assocify/model/database/UserAPI.kt:331]: fixed

**Review by Mai-LinhC — APPROVED** [review]
Perfect thank you !

**Pull request merged**